packages/hardhat/contracts/QuestFactory.sol

```diff
@@ -33,16 +55,26 @@ contract QuestFactory {
             _rewardToken,
             _expireTime,
             aragonGovernAddress,
-            _fundsRecoveryAddress
+            _fundsRecoveryAddress,
+            deposit.token,
+            deposit.amount,
+            msg.sender
         );
+        // Collect deposit from quest creator
+        deposit.collectFrom(msg.sender);
```

Comment:
Maybe put that before `new Quest` happend? I know if it fails it will revert, but looks more easy to understand the requirements, what you do think, make sense?

Reply:
You are right